packages/iocraft/src/handler.rs

```diff
@@ -36,17 +34,65 @@ where
     }
 }
 
-impl<'a, T: 'a> Deref for Handler<'a, T> {
+impl<'a, T: 'a> Deref for HandlerMut<'a, T> {
     type Target = dyn FnMut(T) + Send + Sync + 'a;
 
     fn deref(&self) -> &Self::Target {
-        &self.1
+        self.1.as_ref()
     }
 }
 
-impl<'a, T: 'a> DerefMut for Handler<'a, T> {
+impl<'a, T: 'a> DerefMut for HandlerMut<'a, T> {
     fn deref_mut(&mut self) -> &mut Self::Target {
-        &mut self.1
+        self.1.as_mut()
+    }
+}
+
+/// Immutable variant of [`HandlerMut`]: it lacks ability to mutate captured variables, but can be cloned.
```

Comment:
I think it's important to point out that this can be turned into `HandlerMut`. Maybe say something like...

> Just as `Fn` can be used where an `FnMut` is expected, `Handler` can be used where a `HandlerMut` is expected via `From`/`Into`.

Also, it is probably worth pointing out that components should usually accept properties via `HandlerMut` instead of `Handler`.

Reply:
Tried my best at writing docs for `Handler`, and also included an example. It still is a bit confusing, but at least there is a clear and communicated rationale for the distinction.